be1-go/channel/lao/mod_test.go

```diff
@@ -21,7 +26,115 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-func TestBaseChannel_RollCallOrder(t *testing.T) {
+func TestLAOChannel_Subscribe(t *testing.T) {
+	keypair := generateKeyPair(t)
+
+	fakeHub, err := NewfakeHub(keypair.public, nolog, nil)
+	require.NoError(t, err)
+
+	m := message.Message{MessageID: "0"}
+
+	channel := NewChannel("channel0", fakeHub, m, nolog)
+
+	laoChannel, ok := channel.(*Channel)
+	require.True(t, ok)
+
+	relativePath := filepath.Join("..", "..", "..", "protocol",
+		"examples", "query", "subscribe")
+
+	file := filepath.Join(relativePath, "subscribe.json")
+	buf, err := os.ReadFile(file)
+	require.NoError(t, err)
+
+	var message method.Subscribe
+	err = json.Unmarshal(buf, &message)
+	require.NoError(t, err)
+
+	socket := &fakeSocket{id: "socket"}
+	
+	err = channel.Subscribe(socket, message)
+	require.NoError(t, err)
+
+	require.True(t, laoChannel.sockets.Delete("socket"))
+}
+
+func TestLAOChannel_Unsubscribe(t *testing.T) {
+	keypair := generateKeyPair(t)
+
+	fakeHub, err := NewfakeHub(keypair.public, nolog, nil)
+	require.NoError(t, err)
+
+	m := message.Message{MessageID: "0"}
+	channel := NewChannel("channel0", fakeHub, m, nolog)
+
+	laoChannel, ok := channel.(*Channel)
+	require.True(t, ok)
+
+	relativePath := filepath.Join("..", "..", "..", "protocol",
+		"examples", "query", "unsubscribe")
+
+	file := filepath.Join(relativePath, "unsubscribe.json")
+	buf, err := os.ReadFile(file)
+	require.NoError(t, err)
+
+	var message method.Unsubscribe
+	err = json.Unmarshal(buf, &message)
+	require.NoError(t, err)
+
+	socket := &fakeSocket{id: "socket"}
+	laoChannel.sockets.Upsert(socket)
+
+	require.NoError(t, channel.Unsubscribe("socket", message))
```

Comment:
Some additional checks you can do: 
- check that the socket has been deleted (can use socket.Delete and check for false)
- check that a second call to unsubscribe returns an error